### DISCUSSION TITLE
Using integration to populate energy dashboard

### DISCUSSION THREAD

**Discussion by LukeMc8** [nathanmarlor/foxess_modbus#82] [Q&A] [answered]
I have a H1 invertor and i'm using the USR-W610 RS485 to LAN modbus addapter in an attempt to integrate my solar setup with home assistant. I tried to use the stealthchesnut integration however a number of the sensors were showing as unavailable, I have just installed this integration and all sensors appear to working perfectly. My next task is to use this information to populate the energy dashboard so I need to create the grid daily, feed in daily etc sensors from the sensors created by this integration. I'm looking for some pointers on how to do this or am I going about this all wrong?

**Comment by canton7** [comment, marked as answer]
Grab the latest beta release -- we just added support for a bunch of 'Total' senors that the energy dashboard needs.

* For Grid Consumption, select Grid Consumption Total
* For Return to Grid, select Feed-in Total
* Solar Production:
    * If you have a single string, you can select Solar Generation Total
    * If you have multiple strings and want to show them separately, you'll need to create riemann integral helpers for PV1 Power and PV2 Power, and use those
* For Energy going into battery, select Battery Charge Total
* For Energy coming out of battery, select Battery Discharge Total

**Reply by nathanmarlor** [reply]
I'll add this to the readme when I get chance 🔥

**Reply by canton7** [reply]
@nathanmarlor I was wondering... Is it possible to include integration sensors for PV1/PV2 as part of foxess_modbus? Reading [the source](https://github.com/home-assistant/core/blob/dev/homeassistant/components/integration/sensor.py), it might be as easy as just creating some `IntegrationSensor` instances...

**Reply by nathanmarlor** [reply]
@canton7 hadn't even considered it but it'd make perfect sense and would ease setting up the energy dashboard even further 😁 One to add the weekend to-do list? 👍

**Reply by canton7** [reply]
I guess people can always disable them if they don't want them! Yeah I think that would be well worth doing, if it's possible. I probably won't have much time this weekend, but I'll pick it up next week sometime if you don't get there first.

**Reply by nathanmarlor** [reply]
I wonder how much HA hacking would be involved to automate the setup of the energy dashboard... 🤪

**Reply by nathanmarlor** [reply]
Also thought - and we're talking real gold plating here - but we could offer a drop down of extended / recommended / minimal sensor sets at configuration time. Perhaps even a set linked to the requirements of the energy dashboard.

**Reply by canton7** [reply]
As in, only fetch a subset of the sensors?

I've been toying around with the idea of noticing when all sensors which are interested in a particular modbus address are disabled, and use that to avoid fetching that address. I guess your configuration could tie in there: it could disable some of the sensors by default, and then the user can still show and re-enable them individually if they want?

**Reply by LukeMc8** [reply]
Thanks, I've updated to the beta now and i can see the total sensors. Can i just check that for grid consumption i would chose grid consumption total and not grid consumption today? I need to have a look at the solar generation a little as my setup includes PV1, PV2 and CT2 with CT2 montioring a seperate FOX ESS F5000 inverter. 

Excellent work your doing, if i can help in testing at all let me know.

**Reply by canton7** [reply]
Yep, you want the Total sensors. The Today sensors reset at midnight, which confuses the Energy Dashboard (you'll get a large negative bar at the start of the day)

**Reply by LukeMc8** [reply]
Cheers, once again excellent job.
